projects/ipms-mock/server.py

```diff
@@ -18,79 +22,84 @@
-            full_message_type = f'{message_type}^{trigger_event}'
+            # Properly extract message_type
+            full_message_type = self.incoming_message['msh']['message_type']
 
             logging.info('Handling message type: %s', full_message_type)
 
-            if message_type == 'ORM' and trigger_event == 'O01':
-                response = self.handle_orm()
-            elif message_type == 'ORU' and trigger_event == 'R01':
-                response = self.handle_oru()
-            elif message_type == 'ADT' and (trigger_event in ['A01', 'A02', 'A03']):
-                response = self.handle_adt()
+            if full_message_type == 'ORM^O01':
+                response = handle_orm_o01(self.incoming_message, datastore)
+            elif full_message_type == 'ADT^A04':
+                response = handle_adt_a04(self.incoming_message, datastore)
             else:
                 raise UnsupportedMessageType(
                     f"Unsupported message type: {full_message_type}")
         except Exception as e:
             logging.error("Error handling message: %s", str(e))
-            response = self.create_error_response(str(e))
+            response = create_error_response(str(e))
         return response
 
-    def handle_orm(self):
-        # Process ORM message and prepare response
-        logging.debug("Handling ORM message")
-        response = self.create_ack_response()
-        return response.to_mllp()
-
-    def handle_oru(self):
-        # Process ORU message and prepare response
-        logging.debug("Handling ORU message")
-        response = self.create_ack_response()
-        return response.to_mllp()
-
-    def handle_adt(self):
-        # Process ADT message and prepare response
-        logging.debug("Handling ADT message")
-        response = self.create_ack_response()
-        return response.to_mllp()
-
-    def create_ack_response(self):
-        # Common method to create an ACK response for any message type
-        response = Message("ACK")
-        response.msh.msh_9 = 'ACK'
-        response.msh.msh_10 = uuid.uuid4().hex
-        response.msa.msa_1 = 'AA'  # Acknowledgment code
-        response.msa.msa_2 = self.incoming_message.msh.msh_10
-        return response
-
-    def create_error_response(self, error_msg):
-        logging.warning("Creating error response: %s", error_msg)
-        response = self.create_ack_response()
-        response.msa.msa_1 = 'AE'  # Error acknowledgment code
-        response.err.err_1 = error_msg
-        return response.to_mllp()
-
 if __name__ == '__main__':
     # Define handlers for each message type expected
     handlers = {
         'ORM^O01': (HL7Handler,),
-        'ORU^R01': (HL7Handler,),
-        'ADT^A01': (HL7Handler,),  # Admit a patient
-        'ADT^A02': (HL7Handler,),  # Transfer a patient
-        'ADT^A03': (HL7Handler,)   # Discharge a patient
+        # 'ORU^R01': (HL7Handler,),
```

Comment:
Are these commented messages to be handled later at an advanced phase?

Reply:
@MohamedAliBouhaouala These commented out message handler types can probably be removed. This was added in initially when not knowing the messages types we would need to process. It has since been confirmed by Piotr that the SHR only sends ADT^04 and ORM^O01 messages to IPMS.

Unless additional messages need to be catered for in future but Im currently not aware of those